src/wayland/handlers/screencopy.rs

```diff
@@ -1144,8 +1149,30 @@ impl State {
         output: &Output,
     ) -> Option<Vec<(Session, BufferParams)>> {
         let workspace = self.common.shell.active_space_mut(output);
-        if !workspace.pending_buffers.is_empty() {
-            Some(std::mem::take(&mut workspace.pending_buffers))
+        let mut pending_buffers = std::mem::take(&mut workspace.pending_buffers);
+
+        let mut i = 0;
```

Comment:
Looks like the `drain_filter`-pattern? :sweat_smile: 

Could you add a `TODO: drain_filter` comment here? I have not yet given up on this feature becoming stable at some point!

Reply:
Maybe it would be good to have a `utils::drain_filter` function.

Or there's the the micro-crate solution of just depending on https://crates.io/crates/drain_filter_polyfill.

Reply:
yes, I have thought about that. But that should be done in another PR imo.